### PR TITLE
feat(usage): add PDF review pack to exports

### PR DESCRIPTION
**Pull request by rohoswagger**
## Description

Adds `usage_report.pdf` to the usage export archive. The review pack summarizes spend, usage, seat adoption, and attribution by user, model, and flow.

The report uses the same usage rows as `usage_by_user.csv` so totals reconcile. PDF render failures do not block the CSV export.

The change adds ReportLab-based rendering and enterprise branding support, including a wordmark fallback when a custom logo cannot render.

## How Has This Been Tested?

[onyx_usage_report_branded.pdf](https://github.com/user-attachments/files/30954081/onyx_usage_report_branded.pdf)

- Ran repository pre-commit hooks, including type, lint, formatting, lockfile, and secret checks.
- Added unit coverage for usage-report aggregation and branding behavior.
- Added an integration assertion for the PDF export artifact.
- Did not run the full unit or integration suites locally.

## Additional Options

- [ ] [Optional] Please cherry-pick this PR to the latest release version.
- [x] [Optional] Override Linear Check

### CODE REVIEW

File: backend/ee/onyx/server/reporting/usage_report_pdf.py [hunk 1]
```diff
@@ -0,0 +1,617 @@
...
+                ("VALIGN", (0, 0), (-1, -1), "MIDDLE"),
+            ]
+        )
+    )
+    return table
+
+
+def _axis_labels(days: list[str]) -> list[str]:
+    """Keep at most `_MAX_AXIS_LABELS` ticks, blanking the rest."""
+    step = max(1, len(days) // _MAX_AXIS_LABELS)
```
[the remark this answers is not in the document]
Comment (rohoswagger) [reply backend/ee/onyx/server/reporting/usage_report_pdf.py]: Fixed in 3ed7a8abab. _axis_labels now uses ceiling division, and the focused unit test covers 13, 23, and 30-day periods (plus boundary cases) to keep rendered labels at or below 12.

[the remark this answers is not in the document]
Comment (rohoswagger) [reply backend/ee/onyx/server/reporting/usage_report_pdf.py]: Valid. The chart helper now uses a ceiling interval:

`_axis_labels()` computes `step = max(1, (len(days) + _MAX_AXIS_LABELS - 1) // _MAX_AXIS_LABELS)`, so 13, 23, 24, 25, 30, and longer ranges cannot render more than 12 visible labels.

Regression coverage is in `test_axis_labels_never_exceed_the_display_limit()`.

File: backend/ee/onyx/server/reporting/usage_report_data.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    total_input = 0
+    total_output = 0
+    total_cache_read = 0
+    active_emails: set[str] = set()
+
+    for row in rows:
+        for bucket, key in (
+            (by_user, row.email),
+            (by_model, row.model),
+            (by_flow, row.flow),
```
[the remark this answers is not in the document]
Comment (rohoswagger) [reply backend/ee/onyx/server/reporting/usage_report_data.py]: Fixed in 3ed7a8abab. PDF aggregation now maps an empty flow to other, matching the existing usage UI behavior; test_unlabeled_flow_is_grouped_as_other verifies the readable label and preserved spend.

[the remark this answers is not in the document]
Comment (rohoswagger) [reply backend/ee/onyx/server/reporting/usage_report_data.py]: Valid. Untagged flow rows are now normalized before aggregation.

`build_usage_report_data()` groups `by_flow` with `row.flow or UNLABELED_FLOW`, where `UNLABELED_FLOW = "other"`. `test_unlabeled_flow_is_grouped_as_other()` verifies the PDF aggregate uses `other` and keeps spend reconciled.

File: backend/ee/onyx/server/reporting/usage_report_pdf.py [hunk 3]
```diff
@@ -0,0 +1,617 @@
...
+    }
+
+
+def _logo_flowable(branding: ReportBranding) -> Flowable | None:
+    if not branding.logo:
+        return None
+    try:
+        reader = ImageReader(BytesIO(branding.logo))
+        src_w, src_h = reader.getSize()
+    except Exception:
```
[the remark this answers is not in the document]
Comment (rohoswagger) [reply backend/ee/onyx/server/reporting/usage_report_pdf.py:160]: Fixed in a37f8dd15c. _logo_flowable now logs the decoding exception and application name before returning the existing wordmark fallback, so malformed raster uploads remain non-fatal but diagnosable.

[the remark this answers is not in the document]
Comment (rohoswagger) [reply backend/ee/onyx/server/reporting/usage_report_pdf.py:160]: Valid. The PDF renderer now keeps this fallback observable.

`_logo_flowable()` logs the caught exception with `logger.exception(...)` and includes the application name, then returns `None` so the report still renders through the wordmark path. Existing corrupt-logo coverage still verifies PDF generation succeeds.

File: backend/ee/onyx/server/reporting/usage_report_pdf.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rohoswagger) [reply backend/ee/onyx/server/reporting/usage_report_pdf.py]: Valid. This is the same root cause as the other x-axis thread.

`_axis_labels()` now uses the ceiling interval formula, and `test_axis_labels_never_exceed_the_display_limit()` covers the boundary cases that previously exceeded the 12-label cap.

File: docs/usage/usage-reports.md [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+## Build order
+
+Ordered by value per unit of work. The first two are independent of each other.
+
+1. **Seat ledger.** Fix `users.csv` and derive the dormant-seat list. Small
+   change, and it makes every other file joinable.
+2. **Knowledge gap report.** The differentiated artifact.
+3. **Team and agent dimensions** on the cost breakdown. Unblocks chargeback.
+4. **Summary, manifest, and README** on the export.
+5. **PDF review pack.** Needs the aggregates from steps 1 to 3 to exist first.
```
[the remark this answers is not in the document]
Comment (rohoswagger) [reply docs/usage/usage-reports.md]: Fixed in 3ed7a8abab and refined in a37f8dd15c. The document now states that the archive includes the shipped PDF review pack, and the roadmap describes extending that pack as new dimensions become available.

[the remark this answers is not in the document]
Comment (rohoswagger) [reply docs/usage/usage-reports.md]: Valid. The document now matches the shipped artifact.

The current-state section lists `usage_report.pdf` in the generated zip. The build-order section now says to extend the shipped PDF pack as the seat, knowledge-gap, team, and agent aggregates land, instead of gating the PDF pack behind those future steps.

File: backend/ee/onyx/server/reporting/usage_report_branding.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+            "Usage report rendering %s as a wordmark: the configured logo is "
+            "not a raster image ReportLab can draw",
+            name,
+        )
+        return ReportBranding(application_name=name, logo=None)
+
+    if logo is None:
+        try:
+            logo = _FALLBACK_LOGO.read_bytes()
+        except OSError:
```
[the remark this answers is not in the document]
Comment (rohoswagger) [reply backend/ee/onyx/server/reporting/usage_report_branding.py:76]: Valid. The fallback-logo read failure now includes exception details.

`load_report_branding()` uses `logger.exception("Usage report could not read the fallback logo")` while preserving the non-fatal fallback behavior, so operators get the traceback without blocking report generation.

File: backend/ee/onyx/server/reporting/usage_report_branding.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    if stored.mime_type not in _SUPPORTED_LOGO_TYPES:
+        logger.info(
+            "Usage report cannot draw logo %s of type %s", file_id, stored.mime_type
+        )
+        return None
+
+    return stored.data
+
+
+def load_report_branding(file_store: FileStore) -> ReportBranding:
```
Comment (evan-onyx) [inline backend/ee/onyx/server/reporting/usage_report_branding.py:51]: somewhat confused why this is new code, surely we already have logic somewhere that decides on the whitelabeling we use across the product? If not I'd be in favor of refactoring so that's the case

File: backend/ee/onyx/server/reporting/usage_report_pdf.py [hunk 7]
```diff
@@ -0,0 +1,624 @@
...
+class _SeatMeter(Flowable):
+    """Seats in use against seats bought."""
+
+    def __init__(self, active: int, licensed: int, unseated: int, width: float) -> None:
+        super().__init__()
+        self.active, self.licensed, self.unseated = active, licensed, unseated
+        self.width, self.height = width, 54
+
+    def draw(self) -> None:
+        c = self.canv
```
Comment (evan-onyx) [inline backend/ee/onyx/server/reporting/usage_report_pdf.py:226]: I feel like this is pretty hard to read/debug/modify, there probably isn't a significantly better option tho so best we can do is put comments explaining what each section does (or wrapping them in helper functions)

**Review by evan-onyx — APPROVED** [review]

File: backend/ee/onyx/server/reporting/usage_report_branding.py (hunk 6 above)
Comment (rohoswagger) [reply backend/ee/onyx/server/reporting/usage_report_branding.py:51]: I audited the existing enterprise-settings paths. This module already reuses load_runtime_settings plus the canonical get_logo_filename/get_logotype_filename helpers. The only other backend logic is the HTTP API that returns each configured file separately; it does not choose a logo or provide raw bytes for a renderer. This report-specific loader also needs ReportLab MIME validation and a wordmark fallback, so extracting a shared helper now would add a one-consumer abstraction and pull rendering policy into enterprise settings. I am keeping it local; we should extract it when another backend renderer needs the same policy.

File: backend/ee/onyx/server/reporting/usage_report_pdf.py (hunk 7 above)
Comment (rohoswagger) [reply backend/ee/onyx/server/reporting/usage_report_pdf.py:226]: Fixed in 094b3bd1d4. I grouped the TableStyle commands with concise comments for typography, alignment, separators, and spacing. The rendering commands are unchanged, but the visual sections are now easier to inspect and modify.
